### PR TITLE
doc: add server preload and hubble deploy

### PR DESCRIPTION
**Pull request by aroundabout**
## main change 
1. add the server preload.
2. add how to deploy hubble.

## related issue
1. https://github.com/apache/incubator-hugegraph/issues/840

### CODE REVIEW

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 1]
```diff
@@ -32,7 +32,109 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
...
+有三种方式可以部署`hugegraph-hubble`
+- 使用toolchain
+- 源码编译
+- 使用docker
+
+#### 2.1 使用toolchain
+
+首先下载tar包
+
+```
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-hubble.md:46]:
```suggestion
```bash
```

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 2]
```diff
@@ -32,7 +32,109 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
...
+wget https://downloads.apache.org/incubator/hugegraph/1.0.0/apache-hugegraph-toolchain-incubating-{version}.tar.gz
+tar -xvf apache-hugegraph-toolchain-incubating-{version}.tar.gz 
+cd apache-hugegraph-toolchain-incubating-{version}.tar.gz/apache-hugegraph-hubble-incubating-{version}
+```
+
+运行`hubble`
+
+```
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-hubble.md:54]:
```suggestion
```bash
```

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 3]
```diff
@@ -32,7 +32,109 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
...
+bin/start-hubble.sh
+```
+
+随后我们可以看到
+
+```shell
+starting HugeGraphHubble ..............timed out with http status 502
+2023-08-30 20:38:34 [main] [INFO ] o.a.h.HugeGraphHubble [] - Starting HugeGraphHubble v1.0.0 on cpu05 with PID 3422816 (/home/dandelion/apache-hugegraph-toolchain-incubating-1.0.0/apache-hugegraph-hubble-incubating-1.0.0/lib/hubble-be-1.0.0.jar started by dandelion in /home/dandelion/apache-hugegraph-toolchain-incubating-1.0.0/apache-hugegraph-hubble-incubating-1.0.0)
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-hubble.md:62]:
```suggestion
2023-08-30 20:38:34 [main] [INFO ] o.a.h.HugeGraphHubble [] - Starting HugeGraphHubble v1.0.0 on cpu05 with PID xxx (~/apache-hugegraph-toolchain-incubating-1.0.0/apache-hugegraph-hubble-incubating-1.0.0/lib/hubble-be-1.0.0.jar started by $USER in ~/apache-hugegraph-toolchain-incubating-1.0.0/apache-hugegraph-hubble-incubating-1.0.0)
```

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 4]
```diff
@@ -32,7 +32,109 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
...
+2023-08-30 20:38:37 [main] [INFO ] o.a.c.c.StandardService [] - Starting service [Tomcat]
+2023-08-30 20:38:37 [main] [INFO ] o.a.c.c.StandardEngine [] - Starting Servlet engine: [Apache Tomcat/9.0.24]
+2023-08-30 20:38:37 [main] [INFO ] o.a.c.c.C.[.[.[/] [] - Initializing Spring embedded WebApplicationContext
+2023-08-30 20:38:37 [main] [INFO ] c.z.h.HikariDataSource [] - hugegraph-hubble-HikariCP - Starting...
+2023-08-30 20:38:38 [main] [INFO ] c.z.h.HikariDataSource [] - hugegraph-hubble-HikariCP - Start completed.
+2023-08-30 20:38:41 [main] [INFO ] o.a.c.h.Http11NioProtocol [] - Starting ProtocolHandler ["http-nio-0.0.0.0-8088"]
+2023-08-30 20:38:41 [main] [INFO ] o.a.h.HugeGraphHubble [] - Started HugeGraphHubble in 7.379 seconds (JVM running for 8.499)
+```
+
+然后使用浏览器访问 `ip:8088` 看到`hubble`页面, 也可以通过`bin/stop-hubble.sh`停止服务
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-hubble.md:85]:
```suggestion
然后使用浏览器访问 `ip:8088` 可看到`hubble`页面, 通过`bin/stop-hubble.sh`则可以停止服务
```

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 5]
```diff
@@ -32,7 +32,109 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
...
+
+#### 2.2 源码编译
+
+需要用户提前安装`yarn==16.x`
+
+下载toolchain源码包
+
+```shell
+git clone https://github.com/apache/incubator-hugegraph-toolchain.git
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-hubble.md:94]:
```suggestion
git clone https://github.com/apache/hugegraph-toolchain.git
```

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 6]
```diff
@@ -32,7 +32,109 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
...
+
+```shell
+cd incubator-hugegraph-toolchain/hugegraph-hubble
+mvn -e compile package -Dmaven.javadoc.skip=true -Dmaven.test.skip=true -ntp
+cd apache-hugegraph-hubble-incubating*
+```
+
+启动`hubble`
+
+```
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-hubble.md:107]:
```suggestion
```bash
```

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 7]
```diff
@@ -32,7 +32,109 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
...
+bin/start-hubble.sh -d
+```
+
+#### 2.3 使用Docker
+
+> 注意，如果使用docker启动hubble，如果hubble和server位于同一宿主机，在后续设置graph的hostname的时候请不要直接设置localhost或127.0.0.1，这将指向hubble容器内部而非宿主机，导致无法连接到graph
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-hubble.md:113]:
```suggestion
> **特别注意**: 如果使用 docker 启动 hubble，且 hubble 和 server 位于同一宿主机，在后续 hubble 页面中设置 graph 的 hostname 的时候请不要直接设置 `localhost/127.0.0.1`，这将指向 hubble 容器内部而非宿主机，导致无法连接到 server
```

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 8]
```diff
@@ -32,7 +32,109 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
 
 对于需要遍历全图的Gremlin任务，索引的创建与重建等耗时较长的异步任务，平台提供相应的任务管理功能，实现异步任务的统一的管理与结果查看。
 
-### 2	平台使用流程
+#### 2 部署
+
+有三种方式可以部署`hugegraph-hubble`
+- 使用toolchain
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-hubble.md:38]:
```suggestion
- 下载 toolchain 二进制包
```

同修改后续相关部分文字 (include EN)

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 9]
```diff
@@ -32,7 +32,109 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
...
+- 源码编译
+- 使用docker
+
+#### 2.1 使用toolchain
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-hubble.md:42]: ditto

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 10]
```diff
@@ -32,7 +32,109 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
...
+2023-08-30 20:38:41 [main] [INFO ] o.a.h.HugeGraphHubble [] - Started HugeGraphHubble in 7.379 seconds (JVM running for 8.499)
+flection.CachedClass
+WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
+WARNING: All illegal access operations will be denied in a future release
+2023-08-30 20:38:37 [main] [INFO ] o.a.c.h.Http11NioProtocol [] - Initializing ProtocolHandler ["http-nio-0.0.0.0-8088"]
+2023-08-30 20:38:37 [main] [INFO ] o.a.c.c.StandardService [] - Starting service [Tomcat]
+2023-08-30 20:38:37 [main] [INFO ] o.a.c.c.StandardEngine [] - Starting Servlet engine: [Apache Tomcat/9.0.24]
+2023-08-30 20:38:37 [main] [INFO ] o.a.c.c.C.[.[.[/] [] - Initializing Spring embedded WebApplicationContext
+2023-08-30 20:38:37 [main] [INFO ] c.z.h.HikariDataSource [] - hugegraph-hubble-HikariCP - Starting...
+2023-08-30 20:38:38 [main] [INFO ] c.z.h.HikariDataSource [] - hugegraph-hubble-HikariCP - Start completed.
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-hubble.md:80]: could reduce useless log print 🖨️

File: content/cn/docs/quickstart/hugegraph-server.md [hunk 11]
```diff
@@ -518,4 +525,67 @@ schema = graph.schema()
 
 代表创建示例图成功。
 
-> 使用 IntelliJ IDEA 在启动 Server 时创建示例图的流程类似，不再赘述。
\ No newline at end of file
+> 使用 IntelliJ IDEA 在启动 Server 时创建示例图的流程类似，不再赘述。
+
+
+#### 9.2 启动脚本时使用参数
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-server.md:531]:
```suggestion
#### 9.2 启动脚本时指定参数
```

File: content/cn/docs/quickstart/hugegraph-server.md [hunk 12]
```diff
@@ -518,4 +525,67 @@ schema = graph.schema()
...
+
+在启动脚本后加入`-p true`
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-server.md:533]:
```suggestion
在脚本启动时候携带 `-p true` 参数
```

File: content/cn/docs/quickstart/hugegraph-server.md [hunk 13]
```diff
@@ -518,4 +525,67 @@ schema = graph.schema()
...
+```
+
+并且使用 RESTful API 请求 `HugeGraphServer` 得到如下结果：
+
+```javascript
+> curl "http://localhost:8080/graphs/hugegraph/graph/vertices" | gunzip
+
+  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                 Dload  Upload   Total   Spent    Left  Speed
+100   222  100   222    0     0   4625      0 --:--:-- --:--:-- --:--:--  4625
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-server.md:548]: no need print progress info

File: content/cn/docs/quickstart/hugegraph-server.md [hunk 14]
```diff
@@ -518,4 +525,67 @@ schema = graph.schema()
...
+    使用命令 `docker-compose up -d` 启动容器
+
+使用 RESTful API 请求 `HugeGraphServer` 得到如下结果：
+
+```javascript
+> curl "http://localhost:18080/graphs/hugegraph/graph/vertices" | gunzip
+
+% Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
+                                Dload  Upload   Total   Spent    Left  Speed
+100   222  100   222    0     0   4625      0 --:--:-- --:--:-- --:--:--  4625
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-server.md:588]: same

**Review by imbajin — COMMENTED** [review]
fix CN first, then sync EN doc later

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 15]
```diff
@@ -32,7 +32,93 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
...
+2023-08-30 20:38:41 [main] [INFO ] o.a.h.HugeGraphHubble [] - Started HugeGraphHubble in 7.379 seconds (JVM running for 8.499)
+```
+
+然后使用浏览器访问 `ip:8088` 可看到`hubble`页面, 通过`bin/stop-hubble.sh`则可以停止服务
+
+#### 2.2 源码编译
+
+需要用户提前安装`yarn==16.x`
+
+下载toolchain源码包
```
Comment (javeme) [inline content/cn/docs/quickstart/hugegraph-hubble.md:75]: 我们可以先说明一下："hubble 在 toolchain 仓库中，首先下载 toolchain 源码包"

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 16]
```diff
@@ -32,7 +32,93 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
...
+2023-08-30 20:38:38 [main] [INFO ] c.z.h.HikariDataSource [] - hugegraph-hubble-HikariCP - Start completed.
+2023-08-30 20:38:41 [main] [INFO ] o.a.c.h.Http11NioProtocol [] - Starting ProtocolHandler ["http-nio-0.0.0.0-8088"]
+2023-08-30 20:38:41 [main] [INFO ] o.a.h.HugeGraphHubble [] - Started HugeGraphHubble in 7.379 seconds (JVM running for 8.499)
+```
+
+然后使用浏览器访问 `ip:8088` 可看到`hubble`页面, 通过`bin/stop-hubble.sh`则可以停止服务
+
+#### 2.2 源码编译
+
+需要用户提前安装`yarn==16.x`
```
Comment (javeme) [inline content/cn/docs/quickstart/hugegraph-hubble.md:73]: 我们也给出一下安装命令？

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 17]
```diff
@@ -32,7 +32,93 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
...
+#### 2 部署
+
+有三种方式可以部署`hugegraph-hubble`
+- 下载 toolchain 二进制包
+- 源码编译
+- 使用docker
+
+#### 2.1 下载 toolchain 二进制包
+
+首先下载tar包
```
Comment (javeme) [inline content/cn/docs/quickstart/hugegraph-hubble.md:44]: 我们可以先说明一下："hubble 在 toolchain 仓库中，首先下载 toolchain 的 tar 包"

File: content/cn/docs/quickstart/hugegraph-server.md [hunk 18]
```diff
@@ -518,4 +525,62 @@ schema = graph.schema()
...
 代表创建示例图成功。
 
-> 使用 IntelliJ IDEA 在启动 Server 时创建示例图的流程类似，不再赘述。
\ No newline at end of file
+> 使用 IntelliJ IDEA 在启动 Server 时创建示例图的流程类似，不再赘述。
+
+
+#### 9.2 启动脚本时指定参数
+
+在脚本启动时候携带 `-p true` 参数
```
Comment (javeme) [inline content/cn/docs/quickstart/hugegraph-server.md:533]: "...携带 `-p true` 参数表示xx"

**Review by javeme — COMMENTED** [review]
nice improvement

**Review by javeme — APPROVED** [review]
LGTM

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 19]
```diff
@@ -32,7 +32,115 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
...
+需要用户提前安装`nodejs==16.x` 与 `yarn`
+
+```bash
+apt install curl build-essential
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | bash
+source ~/.bashrc
+nvm install 16
+```
+
+然后确认安装版本是否为16.x
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-hubble.md:82]:
```suggestion
然后确认安装版本是否为 `16.x` (请注意过高的 Node 版本会产生冲突)
```

File: content/cn/docs/quickstart/hugegraph-hubble.md [hunk 20]
```diff
@@ -32,7 +32,115 @@ HugeGraph是一款面向分析型，支持批量操作的图数据库系统，
...
+npm install -g yarn
+```
+
+下载toolchain源码包
+
+```shell
+git clone https://github.com/apache/hugegraph-toolchain.git
+```
+
+编译`hubble`
```
Comment (imbajin) [inline content/cn/docs/quickstart/hugegraph-hubble.md:100]:
```suggestion
编译`hubble`, 它依赖 loader 和 client, 编译时需提前构建这些依赖 (后续可跳)
```

**Review by imbajin — COMMENTED** [review]
sync the EN doc, and **batch apply** the suggestions (rather than apply them one by one)

image

File: content/en/docs/quickstart/hugegraph-hubble.md [hunk 21]
```diff
@@ -32,7 +32,116 @@ By inputting the graph traversal language Gremlin, high-performance general anal
...
+2023-08-30 20:38:38 [main] [INFO ] c.z.h.HikariDataSource [] - hugegraph-hubble-HikariCP - Start completed.
+2023-08-30 20:38:41 [main] [INFO ] o.a.c.h.Http11NioProtocol [] - Starting ProtocolHandler ["http-nio-0.0.0.0-8088"]
+2023-08-30 20:38:41 [main] [INFO ] o.a.h.HugeGraphHubble [] - Started HugeGraphHubble in 7.379 seconds (JVM running for 8.499)
+```
+
+Then use a web browser to access `ip:8088` and you can see the `Hubble` page. You can stop the service using bin/stop-hubble.sh.
+
+#### 2.2 Source code compilation
+
+User should install `nodejs==16.x` and `yarn` firstly.
```
Comment (aroundabout) [inline content/en/docs/quickstart/hugegraph-hubble.md:73]:
```suggestion
**Note**: Compiling Hubble requires the user's local environment to have Node.js V16.x and yarn installed.
```

File: content/en/docs/quickstart/hugegraph-hubble.md [hunk 22]
```diff
@@ -32,7 +32,116 @@ By inputting the graph traversal language Gremlin, high-performance general anal
...
+npm install -g yarn
+```
+
+Download the toolchain source code.
+
+```shell
+git clone https://github.com/apache/hugegraph-toolchain.git
+```
+
+Compile `hubble`
```
Comment (aroundabout) [inline content/en/docs/quickstart/hugegraph-hubble.md:100]:
```suggestion
Compile `hubble`. It depends on the loader and client, so you need to build these dependencies in advance during the compilation process (you can skip this step later).
```

File: content/en/docs/quickstart/hugegraph-hubble.md [hunk 23]
```diff
@@ -32,7 +32,116 @@ By inputting the graph traversal language Gremlin, high-performance general anal
...
+User should install `nodejs==16.x` and `yarn` firstly.
+
+```bash
+apt install curl build-essential
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.1/install.sh | bash
+source ~/.bashrc
+nvm install 16
+```
+
+check the version is 16.x or not
```
Comment (aroundabout) [inline content/en/docs/quickstart/hugegraph-hubble.md:82]:
```suggestion
Then, verify that the installed Node.js version is 16.x (please note that higher Node versions may cause conflicts).
```

**Review by imbajin — APPROVED** [review]
LGTM, THX

**Pull request merged**